src/super_gradients/training/sg_trainer/sg_trainer.py

```diff
@@ -471,6 +475,9 @@ def _train_epoch(self, context: PhaseContext, silent_mode: bool = False) -> tupl
             context.update_context(loss_avg_meter=loss_avg_meter, metrics_compute_fn=self.train_metrics)
 
             for batch_idx, batch_items in enumerate(progress_bar_train_loader):
+                if self.max_train_batches is not None and self.max_train_batches <= batch_idx:
```

Comment:
Perhaps we can use `expected_iterations` here instead of checking again whether self.max_train_batches is not None. Think it would make code a little bit less repeatnig

Reply:
Good point, thx!